frontend/src/metabase/dashboard/actions/metadata.js

```diff
@@ -1,16 +1,26 @@
-import { loadMetadataForQueries } from "metabase/redux/metadata";
+import Questions from "metabase/entities/questions";
 import { getMetadata } from "metabase/selectors/metadata";
-
+import { loadMetadataForQueries } from "metabase/redux/metadata";
+import { getLinkTargets } from "metabase/lib/click-behavior";
 import Question from "metabase-lib/Question";
-
 import { isVirtualDashCard } from "../utils";
 
-export const loadMetadataForDashboard = dashCards => (dispatch, getState) => {
+export const loadMetadataForDashboard = dashCards => async dispatch => {
+  const cards = dashCards
+    .filter(dc => !isVirtualDashCard(dc)) // exclude text cards
+    .flatMap(dc => [dc.card].concat(dc.series || []));
+
+  await Promise.all([
+    dispatch(loadMetadataForCards(cards)),
```

Comment:
Extracted loading metadata for cards in a function so I can call it for cards from click actions